### PR TITLE
feat: company api endpoints

### PR DESCRIPTION
**Pull request by AasmundN**
Adds Nuxt server routes to modify companies in the database.

# Changes

- `GET /api/company` returns all companies from the database. If the query parameter `companyUID` is set it will return only the specified company.
- `POST /api/company` is only accessible by administrators and creates a new company in the database. It accepts a body of the format:
```ts
{
   description?: string
   logo?: string
   name: string
   type: 'partner' | 'sponsor' | 'mainPartner'
   webpage: string
}
```
- `PUT /api/company` updates a specified company in the database. The endpoint is accessible by administrators and company administrators (that is a user with type _company_ and with _companyUID_ set to the company that is to be modified). The endpoint accepts the same body as `POST /api/company` with one addition; the `companyUID` key specifies which company to modify.
- `DELETE /api/company` is only accessible by administrators and deletes a company from the database. The `companyUID` key of the request body specified which company to delete.
- Move all script tags to top of component files.

### CODE REVIEW

File: server/api/company/index.put.ts [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+  // the company does not exist in db
+  if (!data)
+    throw createError({
+      statusCode: 404,
+      statusMessage: 'Company not found',
+    })
+
+  // only admins can change company types
+  if (!isAdmin) updatedData.type = data[companyUID].type
```
Comment (ipeglin) [inline server/api/company/index.put.ts:40]: Haha cheeky, but great :)

**Review by ipeglin — APPROVED** [review]
LGTM! But maybe fix the typo

File: server/api/company/index.delete.ts [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    throw createError({
+      statusCode: 401,
+      statusMessage: 'User not authorized',
+    })
+
+  // get request body
+  const { companyUID } = await readBody(event)
+
+  // reference to companies
+  const companiesRed = db.ref('companies')
```
[the remark this answers is not in the document]
Comment (AasmundN) [reply server/api/company/index.delete.ts:18]: Yes, will fix

File: server/api/company/index.put.ts (hunk 1 above)
Comment (AasmundN) [reply server/api/company/index.put.ts:40]: Haha yeah. The more I think about it the more it doesn't make sense. For some reason I thought I had to fetch from the database to see if it existed. I'll refactor this part before merging.

**Review by ipeglin — APPROVED** [review]
LGTM

**Pull request merged**